### PR TITLE
extension: Separate multicluster chart and binary

### LINKED ISSUE #5257: Move multicluster charts to top-level multicluster directory
To follow the structure of a Linkerd extension, the linkerd multicluster charts should be moved into a separate top-level directory called `multicluster`.

### PR DESCRIPTION
**Pull request by Pothulapati**
Fixes #5257

This branch movies mc charts and cli level code to a new
top level directory. None of the logic is changed.

Also, moves some common types into `/pkg` so that they
are accessible both to the main cli and extensions.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: pkg/k8s/resource.go
Comment (alpeb) [inline pkg/k8s/resource.go:1]: WDYT about putting this under a `resource` package and remove all the appearances of the "Resource" word in the functions below? So they'd be called like `resource.Render()`, etc.

**Review by alpeb — COMMENTED** [review]
Nice that you splitted `multicluster.go` up, and also that you extracted the resource stuff out of `uninstall.go`  :+1:  :+1: 

Maybe we can get away without using the `linkerd2` prefix in the chart directories?

Also, you missed updating `setValues()` in `bin/helm-build` :wink:

**Comment by alpeb** [comment]
@Pothulapati You're planning on moving the service mirror controller code in a separate PR?

**Comment by adleong** [comment]
Should we hold this PR until https://github.com/linkerd/linkerd2/pull/5175 merges?

File: multicluster/cmd/root.go
Comment (Pothulapati) [reply multicluster/cmd/root.go:64]: we actually don't read the env variable. So, removing it!

**Comment by alpeb** [comment]
> Should we hold this PR until #5175 merges?

I think we should

File: multicluster/static/templates.go
Comment (Pothulapati) [reply multicluster/static/templates.go:20]: Just realized, we can use the same fn from `getRepoRoot` everywhere. So, doing that.

File: pkg/k8s/resource.go
Comment (Pothulapati) [reply pkg/k8s/resource.go:1]: Hmmm, I like that way but still we can't remove the `Resource` word from the fn's right? How would they named be instead?

**Comment by Pothulapati** [comment]
@alpeb Moved the service-mirror controller code in this PR too!

File: pkg/k8s/resource.go
Comment (alpeb) [reply pkg/k8s/resource.go:1]:
```go
foo := resource.New()
foo = resource.NewNamespaced()
foo.Render()
```

Comment (Pothulapati) [reply pkg/k8s/resource.go]: Done 👍

Comment (alpeb) [reply pkg/k8s/resource.go]: To [avoid stuttering](https://blog.golang.org/package-names) I would get rid of all mentioning of "resource" in that file, besides the package name itself. So instead of `type KubernetesResource` I would put `type Kubernetes` and instead of `func (r KubernetesResource) RenderResource()` it'd be `func (r Kubernetes) Render()`. You can even use an import alias to make things extra clear: `k8sResource "github.com/linkerd/linkerd2/pkg/k8s/resource"`

Comment (Pothulapati) [reply pkg/k8s/resource.go]: Good Suggestion! _(Also great blog suggestion, made things clear)_

> import alias to make things extra clear: k8sResource "github.com/linkerd/linkerd2/pkg/k8s/resource"
Does not seem that useful, so skipped this!

**Review by alpeb — APPROVED** [review]
Thanks @Pothulapati LGTM modulo the merge conflicts :+1:

**Review by kleimkuhler — APPROVED** [review]
Looks good 👍

**Review by adleong — APPROVED** [review]

**Pull request merged**
